src/stk500v2.c

```diff
@@ -3706,11 +3755,20 @@ static void stk500v2_print_parms1(const PROGRAMMER *pgm, const char *p, FILE *fp
         }
         f /= prescale;
         f /= (osc_cmatch + 1);
+        decimals = get_decimals(f);
         f = f_to_kHz_MHz(f, &unit);
-        fmsg_out(fp, "%.3f %s\n", f, unit);
+        fmsg_out(fp, "%.*f %s\n", f, decimals, unit);
```

Comment:
change order of f and decimals arguments (multiple times)

Reply:
yep, 2 times